src/react-dropzone.js

```diff
@@ -100,14 +100,14 @@ export class DropzoneComponent extends React.Component {
    * React 'componentWillUpdate'
    * Update Dropzone options each time the component updates.
    */
-  componentWillUpdate () {
+  componentWillUpdate (nextProps) {
     let djsConfigObj
     let postUrlConfigObj
 
-    djsConfigObj = this.props.djsConfig ? this.props.djsConfig : {}
+    djsConfigObj = nextProps.djsConfig ? nextProps.djsConfig : {}
```

Comment:
Basically this should be like before. But there should be a strict equality comparison for the current and new config, like `nextProps.djsConfig !== this.props.djsConfig`.

Reply:
i don't think componentWillUpdate gets called on the initial render - although you didn't say 'mount' so maybe i'm missing something.

I could see wrapping the whole thing in that strict equality comparison to make sure this doesn't run unless either of those 2 specific props changed - for performance reasons.  Is that what you're looking for?

As far as i can tell there is no reason you would ever want to run the update with the old props like it is doing today.

Reply:
I would use `extend()` to merge them.

I am not completely sure, also because this lifecycle method is deprecated in React 16 because of cases where it is always called.